File: example/android/device/src/main/java/com/qc/device/model/Result.kt
package com.qc.device.model

import com.google.gson.annotations.SerializedName


data class Result<T>(
    @SerializedName("code") val code: Int,
    @SerializedName("message") val message: String? = null,
    @SerializedName("data") val data: T,
)


object  ResultError{
    const val RESULT_OK = 200
    const val PACKAGE_EXCEPTION = 100001
    const val CAMERA_PERMISSION = 100002
    const val CONTACT_PERMISSION = 100003
    const val STORAGE_PERMISSION = 100004
    const val MESSAGE_PERMISSION = 100005
    const val CALENDAR_PERMISSION = 100006
    const val LOCATION_PERMISSION = 100007
    const val CALL_LOG_PERMISSION = 100008
    const val REFERRER_ERROR = 100009
}

File: ios/Classes/model/Result.swift
//
//  Result.swift
//  flutter_device
//
//  Created by atlands on 2023/8/2.
//

class Result<T> {
    let code: Int;
        let message: String?;
        let data: T;
    
    init(code: Int,message: String?, data: T) {
        self.code = code
        self.message = message
        self.data = data
    }
}

struct ResultError {
    static let resultOK = 200
    static let packageException = 100001
    static let cameraPermission = 100002
    static let contactPermission = 100003
    static let storagePermission = 100004
    static let messagePermission = 100005
    static let calendarPermission = 100006
    static let locationPermission = 100007
    static let callLogPermission = 100008
    static let referrerError = 100009
    static let cameraNo = 100010
}

